File: Swift-C++-Testsuite/test/structsAndClasses/Swift/Classes/PolymorphicClass.swift
class Drawable { func draw() {} }

class Point: Drawable {
    init(x: Double, y: Double) {
        self.x = x
        self.y = y
    }
    var x, y: Double
    override func draw() {
        print("Drawing a point")
    }
}
class Line: Drawable {
    init(x1: Double, y1: Double, x2: Double, y2: Double) {
        self.x1 = x1
        self.y1 = y1
        self.x2 = x2
        self.y2 = y2
    }
    var x1, y1, x2, y2: Double
    override func draw() {
        print("Drawing a line")
    }
}
@main
struct MyMain {
    static func main() {
        var point = Point(x: 1, y: 2)
        point.draw()
        var line = Line(x1: 1, y1: 2, x2: 3, y2: 4)
        line.draw()
    }
}

File: Swift-C++-Testsuite/test/structsAndClasses/C++/Classes/PolymorphicClass.cpp
#include "iostream"

class Drawable {
  public:
    virtual void draw() = 0;
};

class Point : public Drawable {
  public:
    double x, y;

    Point (double x, double y): x(x), y(y) {}

    void draw()  override {
      std::cout << "Drawing a point\n";
    }
};

class Line : public Drawable {
  public:
    double x1, y1, x2, y2;
    
    Line(double x1, double y1, double x2, double y2): x1(x1), x2(x2), y1(y1), y2(y2) {
    }

    void draw()  override {
      std::cout << "Drawing a line\n";
    }
};

void callCB(Drawable *F) { F->draw(); }

int main() {
  Point P{1.0, 2.0};
  callCB(&P);
  Line L{1.0, 2.0, 3.0, 4.0};
  callCB(&L);
  return 0;
}
